pkg/tfbridge/internal/schemaconvert/schemaconvert.go

```diff
@@ -0,0 +1,134 @@
+package schemaconvert
+
+import (
+	v1Schema "github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	v2Schema "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+)
+
+func Sdkv2ToV1Type(t v2Schema.ValueType) v1Schema.ValueType {
+	return v1Schema.ValueType(t)
+}
+
+func Sdkv2ToV1SchemaOrResource(elem interface{}) interface{} {
+	switch elem := elem.(type) {
+	case nil:
+		return nil
+	case *v2Schema.Schema:
+		return Sdkv2ToV1Schema(elem)
+	case *v2Schema.Resource:
+		return Sdkv2ToV1Resource(elem)
+	default:
+		contract.Failf("unexpected type %T", elem)
+		return nil
+	}
+}
+
+func Sdkv2ToV1Resource(sch *v2Schema.Resource) *v1Schema.Resource {
+	//nolint:staticcheck
+	if sch.MigrateState != nil {
+		contract.Failf("MigrateState is not supported in conversion")
+	}
+	if sch.StateUpgraders != nil {
+		contract.Failf("StateUpgraders is not supported in conversion")
+	}
+	//nolint:staticcheck
+	if sch.Create != nil || sch.Read != nil || sch.Update != nil || sch.Delete != nil || sch.Exists != nil ||
+		sch.CreateContext != nil || sch.ReadContext != nil || sch.UpdateContext != nil ||
+		sch.DeleteContext != nil || sch.Importer != nil {
+		contract.Failf("runtime methods not supported in conversion")
+	}
+
+	if sch.CustomizeDiff != nil {
+		contract.Failf("CustomizeDiff is not supported in conversion")
+	}
+
+	timeouts := v1Schema.ResourceTimeout{}
+	if sch.Timeouts != nil {
+		timeouts = v1Schema.ResourceTimeout{
+			Create:  sch.Timeouts.Create,
+			Read:    sch.Timeouts.Read,
+			Update:  sch.Timeouts.Update,
+			Delete:  sch.Timeouts.Delete,
+			Default: sch.Timeouts.Default,
+		}
+	}
+	timoutsPtr := &timeouts
+	if sch.Timeouts == nil {
+		timoutsPtr = nil
+	}
```

Comment:
We can make this a lot less awkward:
```suggestion
	var timeouts *v1Schema.ResourceTimeout
	if sch.Timeouts != nil {
		timeouts = &v1Schema.ResourceTimeout{
			Create:  sch.Timeouts.Create,
			Read:    sch.Timeouts.Read,
			Update:  sch.Timeouts.Update,
			Delete:  sch.Timeouts.Delete,
			Default: sch.Timeouts.Default,
		}
	}
```

Reply:
thanks!